typescript/examples/flatbufferswriter/scripts/main.ts

```diff
@@ -0,0 +1,258 @@
+      for (const field of fields) {
+        const { name, offset, type } = field;
+        if (type !== NumericType.FLOAT32) {
+          throw new Error("unsupported numeric types");
+        }
+        const dataFunc = fieldDataFuncs[name];
+        if (dataFunc) {
+          const value = dataFunc({
+            x,
+            y,
+            i,
+            rows: row_count,
+            cols: column_count,
+            time,
+          });
+          view.setFloat32(i + offset, value, true);
+        } else {
+          throw new Error(`missing data func for field: ${name}`);
+        }
+      }
+    }
+  }
+
+  (grid as Grid).data = data;
+}
+
+async function addMessageAtTime(
+  mcapFile: McapWriter,
+  gridChannelId: number,
+  getMessageData: (time: number) => Message["data"],
+  time: bigint,
+): Promise<void> {
+  const message = getMessageData(Number(time));
+  await mcapFile.addMessage({
+    channelId: gridChannelId,
+    sequence: 0,
+    publishTime: time,
+    logTime: time,
+    data: message,
+  });
+}
+
+async function main() {
+  const mcapFilePath = "flatbuffer.mcap";
+  const fileHandle = await open(mcapFilePath, "w");
+  const fileHandleWritable = new FileHandleWritable(fileHandle);
+
+  const mcapFile = new McapWriter({
+    writable: fileHandleWritable,
+    useStatistics: false,
+    useChunks: true,
+    useChunkIndex: true,
+  });
+
+  await mcapFile.start({
+    profile: "",
+    library: "mcap example",
+  });
+  const FrameTransformSchemaBuffer = fs.readFileSync(
+    `${__dirname}/../../flatbuffer/bin/FrameTransform.bfbs`,
+  );
+  const tfSchemaId = await mcapFile.registerSchema({
+    name: "foxglove.FrameTransform",
+    encoding: "flatbuffer",
+    data: FrameTransformSchemaBuffer,
+  });
+
+  const tfChannelId = await mcapFile.registerChannel({
+    schemaId: tfSchemaId,
+    topic: "tf",
+    messageEncoding: "flatbuffer",
+    metadata: new Map(),
+  });
+  const tfJson = {
+    timestamp: { sec: 0, nsec: 0 },
+    parent_frame_id: "base_link",
+    child_frame_id: "sensor",
+    translation: { x: 0, y: 0, z: 1 },
+    rotation: QUAT_IDENTITY,
+  };
+
+  const tfBuilder = new Builder();
+  const tf = buildTfMessage(tfBuilder, tfJson);
+  tfBuilder.finish(tf);
+
+  await mcapFile.addMessage({
+    channelId: tfChannelId,
+    sequence: 0,
+    publishTime: 0n,
+    logTime: 0n,
+    data: tfBuilder.asUint8Array(),
+  });
+
+  const GridSchemaBuffer = fs.readFileSync(`${__dirname}/../../flatbuffer/bin/Grid.bfbs`);
+
+  const gridSchemaId = await mcapFile.registerSchema({
+    name: "foxglove.Grid",
+    encoding: "flatbuffer",
+    data: GridSchemaBuffer,
+  });
+
+  const gridChannelId = await mcapFile.registerChannel({
+    schemaId: gridSchemaId,
+    topic: "grid",
+    messageEncoding: "flatbuffer",
+    metadata: new Map(),
+  });
+
+  const { mcapTimeLength, gridMessageFrequency } = scriptParameters;
+
+  const timeBetweenMessages = 1000 / gridMessageFrequency; // 1s (ms) / frequency
```

Comment:
confused about units here, seems like ms / Hz would give ms*s which is a surprising unit...

Reply:
was trying to do 2 things at once: interval length is 1 / frequency and there are 1000 ms in a second, but yeah the comment wasn't clear. will udpate